gef.py

```diff
@@ -11766,13 +11725,24 @@ def reset_caches(self) -> None:
         return
 
 
+def target_remote_hook():
+    # disable the context until the session has been fully established
+    gef.config["context.enable"] = False
+
+
 def target_remote_posthook():
-    if gef.session.remote_initializing:
-        return
+    conn = gdb.selected_inferior().connection
+    if not isinstance(conn, gdb.RemoteTargetConnection):
+        raise TypeError("Expected type gdb.RemoteTargetConnection")
+    assert is_target_remote_or_extended(conn), "Target is not remote"
+    gef.session.remote = GefRemoteSessionManager(conn)
+
+    # re-enable context
+    gef.config["context.enable"] = True
```

Comment:
If the context was manually disabled by the user before the `target remote` command, you re-enable it anyway, which can be annoying for those who don't want the context

Reply:
Fixed by 14c35bd34a4f67e6a300b7c8093a59259f6fc915